homeassistant/components/iotty/switch.py

```diff
@@ -0,0 +1,181 @@
+        """Handle updated data from the API endpoint."""
+        if not coordinator.last_update_success:
+            return None
+
+        devices = coordinator.data.devices
+        entities = []
+        known_devices: set = config_entry.runtime_data.known_devices
+
+        # Add entities for devices which we've not yet seen
+        for device in devices:
+            if (
+                any(d.device_id == device.device_id for d in known_devices)
+                or device.device_type != LS_DEVICE_TYPE_UID
+            ):
+                continue
+
+            iotty_entity = IottyLightSwitch(
+                coordinator=coordinator,
+                iotty_cloud=coordinator.iotty,
+                iotty_device=LightSwitch(
+                    device.device_id,
+                    device.serial_number,
+                    device.device_type,
+                    device.device_name,
+                ),
+            )
+
+            entities.extend([iotty_entity])
+            known_devices.add(device)
+
+        async_add_entities(entities)
+
+    # Add a subscriber to the coordinator to discover new devices
+    coordinator.async_add_listener(async_update_data)
+
+
+class IottyLightSwitch(SwitchEntity, CoordinatorEntity[IottyDataUpdateCoordinator]):
+    """Haas entity class for iotty LightSwitch."""
+
+    _attr_has_entity_name = True
+    _attr_name = None
+    _attr_entity_category = EntityCategory.CONFIG
+    _attr_device_class = SwitchDeviceClass.SWITCH
+    _iotty_cloud: IottyProxy
+    _iotty_device: LightSwitch
+
+    def __init__(
+        self,
+        coordinator: IottyDataUpdateCoordinator,
+        iotty_cloud: IottyProxy,
+        iotty_device: LightSwitch,
+    ) -> None:
+        """Initialize the LightSwitch device."""
+        super().__init__(coordinator=coordinator)
+
+        _LOGGER.debug(
+            "Creating new SWITCH (%s) %s",
+            iotty_device.device_type,
+            iotty_device.device_id,
+        )
+
+        self._iotty_cloud = iotty_cloud
+        self._iotty_device = iotty_device
+        self._attr_unique_id = iotty_device.device_id
+
+    @property
+    def device_id(self) -> str:
+        """Get the ID of this iotty Device."""
+        return self._iotty_device.device_id
+
+    @property
+    def device_info(self) -> DeviceInfo:
+        """Return the device info."""
+        return DeviceInfo(
+            identifiers={(DOMAIN, cast(str, self._attr_unique_id))},
+            name=self.name,
+            manufacturer="iotty",
+        )
+
+    @property
+    def is_on(self) -> bool:
+        """Return true if the LightSwitch is on."""
+        _LOGGER.debug(
+            "Retrieve device status for %s ? %s",
+            self._iotty_device.device_id,
+            self._iotty_device.is_on,
+        )
+        return self._iotty_device.is_on
+
+    @property
+    def name(self) -> str:
+        """Get the name of this iotty Device."""
+        return self._iotty_device.name
```

Comment:
This is colliding with the `Entity.name` property, please remove it.

Reply:
Done